edx_rest_api_client/client.py

```diff
@@ -81,11 +107,53 @@ def get_oauth_access_token(url, client_id, client_secret, token_type='jwt', gran
     except KeyError:
         raise requests.RequestException(response=response)
 
-    expires_at = now + timedelta(seconds=expires_in)
+    expires_at = now + datetime.timedelta(seconds=expires_in)
 
     return access_token, expires_at
 
 
+# Consider tokens that expire in 5 seconds as already expired
+ACCESS_TOKEN_EXPIRED_THRESHOLD = 5
+
+
+def get_and_cache_oauth_access_token(url, client_id, client_secret, token_type='jwt', grant_type='client_credentials',
+                                     refresh_token=None):
+    """ Retrieves OAuth 2.0 access token using the given grant type, if one is not already cached.
+
+    See ``get_oauth_access_token`` for usage details.
+
+    Returns:
+        tuple: Tuple containing (access token string, expiration datetime).
+
+    """
```

Comment:
I think it would be useful here to describe the caching expiration behavior. I have a sense of it from the code but I think writing out how we would like it to behave would help too.

Reply:
@dianakhuang: I updated the comments.  Let me know if it seems ok.  If not, I can update in the follow-up PR to add the setting.